mep/people/management/commands/export_locations.py

```diff
@@ -0,0 +1,119 @@
+
+"""
+
+from collections import OrderedDict
+from django.db.models import Prefetch
+from mep.common.management.export import BaseExport
+from mep.common.templatetags.mep_tags import domain
+from mep.common.utils import absolutize_url
+from mep.people.models import Location, Person
+from mep.accounts.models import Account, Address
+from mep.accounts.partial_date import DatePrecisionField
+
+
+class Command(BaseExport):
+    """Export member data."""
+
+    help = __doc__
+
+    model = Address
+
+    csv_fields = [
+        "id",  # including "id" to store slug for exports,
+        # given not all exported entities have a URI
+        "member_id",  # member slug
+        "member_uri",
+        "care_of_person_id",  # person slug
+        "street_address",
+        "postal_code",
+        "city",
+        "arrrondissement",
+        "country",
+        "start_date",
+        "end_date",
+        "start_date_precision",
+        "end_date_precision",
+        "longitude",
+        "latitude",
+    ]
+
+    def get_queryset(self):
+        """
+        custom filter needed to return person-address combos,
+        so we can pass a one object per row to `get_object_data`
+        """
+        addresses = Address.objects.prefetch_related(
+            Prefetch("account"),
+            Prefetch("person"),
+            Prefetch("location"),
+        )
+        res = []
+        for addr in addresses.all():
+            persons = [addr.person] if addr.person else addr.account.persons.all()
+            for person in persons:
+                res.append((person, addr))
+        return res
+
+    def get_base_filename(self):
+        """set the filename to 'locations.csv'"""
+        return "locations"
+
+    def get_object_data(self, obj):
+        """
+        Generate dictionary of data to export for a single
+        :class:`~mep.people.models.Person`
+        """
+        person, addr = obj
+        loc = addr.location
+
+        # format id
+        id_parts = [
+            person.slug,
+            loc.street_address,
+            loc.city,
+            addr.start_date.isoformat() if addr.start_date else None,
+            addr.end_date.isoformat() if addr.end_date else None,
+        ]
+        id = "-".join(x for x in id_parts if x).lower()
+
+        # required properties
+        return dict(
+            # Member
+            id=id,
+            member_id=person.slug,
+            member_uri=absolutize_url(person.get_absolute_url()),
+            # Address data
+            start_date=addr.start_date.isoformat() if addr.start_date else None,
+            end_date=addr.end_date.isoformat() if addr.end_date else None,
+            start_date_precision=serialize_precision(addr.start_date_precision)
+            if addr.start_date_precision
+            else None,
+            end_date_precision=serialize_precision(addr.end_date_precision)
+            if addr.end_date_precision
+            else None,
```

Comment:
should be able to just use this:

```suggestion
            start_date=addr.partial_start_date,
            end_date=addr.partial_end_date,
```